src/headless_browser.ts

```diff
@@ -58,82 +67,215 @@ export type ExceptionDetails = { // exists when an error
+    this.socket.onerror = function (e) {
+      webSocketIsDonePromise.resolve();
+    };
+    this.socket.onmessage = (msg) => {
+      // 2nd part of the dirty fix 1
+      const data = JSON.parse(msg.data);
+      if (data.id && data.id === -1) {
+        this.socket!.close();
+      } else {
+        this.handleSocketMessage(msg);
+      }
+    };
+
+    // Enable page notifications, so we can wait for page events, such as when a page has loaded
+    await this.sendWebSocketMessage("Page.enable");
+  }
+
+  /**
+   * Asserts a given url matches the current
+   *
+   * @param expectedUrl - The expected url, eg `https://google.com/hello`
+   */
+  public async assertUrlIs(expectedUrl: string): Promise<void> {
+    // There's a whole bunch of other data it responds with, but we only care about documentURL
+    const res = await this.sendWebSocketMessage("DOM.getDocument") as {
+      root: {
+        documentURL: string;
+      };
+    };
+    const actualUrl = res.root.documentURL;
+    assertEquals(actualUrl, expectedUrl);
+  }
+
+  /**
+   * Check if the given text exists on the dom
+   *
+   * @param text - The text to check for
+   */
+  public async assertSee(text: string): Promise<void> {
+    const command = `document.body.innerText.indexOf('${text}') >= 0`;
+    const res = await this.sendWebSocketMessage("Runtime.evaluate", {
+      expression: command,
+    });
+    this.checkForErrorResult((res as DOMOutput), command);
+    // Tried and tested, and `result` is `{result: { type: "boolean", value: false } }`
+    const exists = ((res as DOMOutput).result as SuccessResult).value;
+    assertEquals(exists, true);
+  }
+
+  /**
+   * Go to a specific page
+   *
+   * @param urlToVisit - The page to go to
+   */
+  public async goTo(urlToVisit: string) {
+    const notificationPromise = this
+      .notification_resolvables["Page.loadEventFired"] = deferred();
+    const res = await this.sendWebSocketMessage("Page.navigate", {
+      url: urlToVisit,
+    }) as {
+      frameId: string;
+      loaderId: string;
+      errorText?: string; // Only present when an error occurred, eg page doesn't exist
+    };
+    await notificationPromise;
+    if (res.errorText) {
+      //await this.done()
+      throw new Error(
+        `${res.errorText}: Error for navigating to page "${urlToVisit}"`,
+      );
+    }
   }
 
   /**
    * Clicks a button with the given selector
-   * Will not allow clicking links, you can check
-   * a link works by just checking the href
    *
    *     await this.click("#username");
    *     await this.click('button[type="submit"]')
    *
    * @param selector - The tag name, id or class
    */
   public async click(selector: string): Promise<void> {
-    if (selector.indexOf("href") !== -1) {
-      throw new Error(
-        "Clicking links is not supported. You can check a link works by getting the `href` attribute",
-      );
-    }
     const command = `document.querySelector('${selector}').click()`;
```

Comment:
similar to this... it could be helpful to have an `eval` function of something 

```
Sinco.eval(() => document.querySelector(...).children());
```

Reply:
I agree, good shout! - i think this could be a separate issue to tackle though?

Reply:
agreed :)